### DISCUSSION TITLE
Variable wheel sha for identical sources

### DISCUSSION THREAD

**Discussion by radugrosu** [PyO3/maturin#981] [Q&A]
I am building a wheel with:
`maturin build -m Cargo.toml --no-sdist --release --strip --manylinux 2_24 -i /opt/python/cp38-cp38/bin/python --target x86_64-unknown-linux-gnu` inside the `ghcr.io/pyo3/maturin` container.
The problem is: the wheel sha's are different for two consecutive runs (which makes my github workflows fail, etc).

What is the reason for this? Is there any way to avoid this?
Thank you :pray:

**Comment by aviramha** [comment]
Strip the binary, it probably contains paths/compilation strings that change on each run.

**Reply by radugrosu** [reply]
If I simply re-execute the command I get different sha's for the resulting wheel.

**Reply by radugrosu** [reply]
Also, the output of `strings` on the different wheels is identical.

**Reply by radugrosu** [reply]
The wheel is possibly including some extraneous bit of information, such as a timestamp - is there a way to have maturin not include that?

**Reply by aviramha** [reply]
it's a zip format thing

**Comment by messense** [comment]
We did not put any effort in making `maturin build` reproducible in the past, and obviously we rely on `cargo build` to be able to do reproducible build.

Feel free to experiment if you want it.

**Reply by radugrosu** [reply]
Hi @messense , thanks  for your answer. I don't think it's got to do anything with cargo: all the files inside the wheel are identical between builds. It's the wheel itself that for some strange reason includes an unwanted piece of metadata (probably a timestamp). I was hoping there was a way to make it stop doing that.
